src/workers/newNotificationV2Mail.ts

```diff
@@ -0,0 +1,702 @@
+  },
+  community_picks_succeeded: async (con, users, notification) => {
+    const post = (await con
+      .getRepository(Post)
+      .findOneBy({ id: notification.referenceId })) as ArticlePost;
+    const submission = await con
+      .getRepository(Submission)
+      .findOneBy({ url: post.url, userId: users[0].id });
+    return {
+      static: {
+        submitted_at: formatMailDate(submission.createdAt),
+        post_title: truncatePostToTweet(post),
+        post_image: post.image || pickImageUrl(post),
+        article_link: post.url,
+        discussion_link: addNotificationEmailUtm(
+          notification.targetUrl,
+          notification.type,
+        ),
+      },
+      personalized: users.map((user) => ({
+        first_name: getFirstName(user.name),
+      })),
+    };
+  },
+  community_picks_granted: async (con, users) => {
+    return {
+      static: {},
+      personalized: users.map((user) => ({
+        first_name: getFirstName(user.name),
+      })),
+    };
+  },
+  article_picked: async (con, users, notification, attachments) => {
+    const att = attachments[0];
+    return {
+      static: {
+        post_title: truncatePostToTweet(att),
+        post_image: att.image,
+        discussion_link: addNotificationEmailUtm(
+          notification.targetUrl,
+          notification.type,
+        ),
+      },
+    };
+  },
+  article_new_comment: async (con, users, notification) => {
+    const comment = await con.getRepository(Comment).findOne({
+      where: { id: notification.referenceId },
+      relations: ['post', 'user'],
+    });
+    if (!comment) {
+      return;
+    }
+    const [commenter, post] = await Promise.all([comment.user, comment.post]);
+    return {
+      static: {
+        profile_image: commenter.image,
+        full_name: commenter.name,
+        post_title: truncatePostToTweet(post),
+        post_image: (post as ArticlePost).image || pickImageUrl(post),
+        new_comment: notification.description,
+        discussion_link: addNotificationEmailUtm(
+          notification.targetUrl,
+          notification.type,
+        ),
+        user_reputation: commenter.reputation,
+      },
+    };
+  },
+  article_upvote_milestone: async (con, users, notification, attachments) => {
+    const att = attachments[0];
+    return {
+      static: {
+        post_title: truncatePostToTweet(att),
+        post_image: att.image,
+        discussion_link: addNotificationEmailUtm(
+          notification.targetUrl,
+          notification.type,
+        ),
+        upvotes: notification.uniqueKey,
+        upvote_title: basicHtmlStrip(notification.title),
+      },
+    };
+  },
+  article_report_approved: async (con, users, notification) => {
+    const post = await con
+      .getRepository(Post)
+      .findOneBy({ id: notification.referenceId });
+    return {
+      static: {
+        post_title: truncatePostToTweet(post),
+        post_image: (post as ArticlePost).image || pickImageUrl(post),
```

Comment:
I know it's existing, wondering if this could also have side effect for video/collection etc?

Reply:
As long as image is defined, it's be ok. It's just a synthetic sugar for typing issue